packages/lodestar/src/sync/utils/blocks.ts

```diff
@@ -83,12 +84,15 @@ export async function getBlockRange(
             +`from peer ${peer.toB58String()}`);
           await sleep(1000);
           //if failed to obtain blocks, try in next round on another peer
-          return chunk;
+          return {chunk, peer};
         }
       })
-    )).filter((chunk) => chunk !== null);
+    )).filter((chunkPeer) => chunkPeer !== null);
+    chunks = chunkPeers.map((chunkPeer) => chunkPeer.chunk);
+    const badPeers = chunkPeers.map((chunkPeer) => chunkPeer.peer);
+    peers = peers.filter((peer) => !badPeers.includes(peer));
```

Comment:
I don't think we should handle this here, those thing should be sorted out once we get peer scoring.
And we probably shouldn't use peers with closed connection (that should be filtered when obtaining peer list)